phylanx/execution_tree/compiler/compiler.hpp

```diff
@@ -293,9 +323,9 @@ namespace phylanx { namespace execution_tree { namespace compiler
     };
 
     ///////////////////////////////////////////////////////////////////////////
-    typedef hpx::util::function_nonser<function(
-            std::list<function> const&, std::string const&, std::string const&
-        )> compiled_function;
+    using compiled_function = hpx::util::function_nonser<function(
+            std::list<function> const&, primitive_name_parts, std::string const&
+        )>;
 
```

Comment:
From what I can see, most of the compiler functions don't modify the name_parts object. Wouldn't it be better to generally pass it by `const&` and copy it explicitly if it needs to be modified?